src/components/CityListScreen.js

```diff
@@ -1,25 +1,75 @@
-import React, { Component } from 'react';
 import { FlatList, View, StyleSheet } from 'react-native';
 import { autobind } from 'core-decorators';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import * as cityService from '../services/city';
+import { connect } from 'react-redux';
+import _ from 'lodash';
 import CityListItem from './CityListItem';
+import { getCities } from '../redux/selectors';
+import { fetchCurrentWeatherForCity } from '../redux/actions';
 
 class CityListScreen extends Component {
   static propTypes = {
     navigation: PropTypes.object,
+    cities: PropTypes.array,
+    fetchCurrentWeatherForCityAction: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      refreshing: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchWeatherForCities();
+  }
+
+  @autobind
+  handleRefresh() {
+    this.setState(
+      {
+        refreshing: true,
+      },
+      () => {
```

Comment:
no need to wrap it in another function. 
`this.setState({ refreshing: true }, this.fetch.WeatherForCities)` is enough.

Reply:
But good job with pull-to-refresh implementation :)